tests/Tests/ORM/Functional/ProxiesLikeEntitiesTest.php

```diff
@@ -83,8 +79,10 @@ public function testEntityWithIdentifier(): void
     {
         $userId             = $this->user->getId();
         $uninitializedProxy = $this->_em->getReference(CmsUser::class, $userId);
-        assert($uninitializedProxy instanceof CmsUserProxy);
-        self::assertInstanceOf(CmsUserProxy::class, $uninitializedProxy);
+        if (! $this->_em->getConfiguration()->isNativeLazyObjectsEnabled()) {
+            assert($uninitializedProxy instanceof CmsUserProxy);
+            self::assertInstanceOf(CmsUserProxy::class, $uninitializedProxy);
```

Comment:
use `$this->assertTrue($this->isUninitializedObject($uninitializedProxy))`, which works for all kind of proxies.